vsi/src/forensics.rs

```diff
@@ -0,0 +1,45 @@
+//! Once a scan (which is local) is completed and all the scan artifacts are uploaded to the VSI Forensics Service,
+//! the results of the forensics investigation must be waited upon and then its results downloaded.
+//!
+//! Only then can the client know which dependencies were discovered for the scan artifacts by the forensics service.
+
+use std::fmt::Display;
+
+/// The VSI Forensics Service returns statuses for tracking on which stage forensic analysis is.
+///
+/// This client only cares about a subset; the rest are informational and can be safely shown to a user to indicate activity.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Status {
+    /// Forensic analysis is pending.
+    Pending,
+    /// Forensic analysis has completed.
+    Finished,
+    /// Forensic analysis has failed.
+    Failed,
+    /// Forensic analysis is in some other status, which can be displayed to the user.
+    /// This status indicates that it is in process.
+    Informational(String),
+}
+
+impl Status {
+    /// Safely parse the string.
+    pub fn parse(input: String) -> Self {
+        match input.as_str() {
```

Comment:
[optional] are these matches case-sensitive? if not, prefer if cast them upper or lower or do case-insensitive match

Reply:
Hmm, they're just an enum server side. So not really case insensitive, but why do the extra work of rendering them lower/upper?

Reply:
These are treated as an enum, and tons of server internals depend on them and are case sensitive: https://github.com/fossas/basis/blob/be26d1cb524f0ebb0128d34eaccc5d6903604a3d/sherlock/internal/models/forensics_pipeline.go#L6-L17

Reply:
Good enough for me, I think we can call that a contract.